packages/strapi-plugin-i18n/admin/src/components/ModalEdit/index.js

```diff
@@ -13,14 +13,29 @@ import SettingsModal from '../SettingsModal';
 
 const ModalEdit = ({ localeToEdit, onClose, locales }) => {
   const { isEditing, editLocale } = useEditLocale();
+  const [shouldUpdateMenu, setShouldUpdateMenu] = useState(false);
+  const { updateMenu } = useGlobalContext();
   const { formatMessage } = useIntl();
   const isOpened = Boolean(localeToEdit);
 
   const handleSubmit = ({ displayName, isDefault }) => {
     const id = localeToEdit.id;
     const name = displayName || localeToEdit.code;
 
-    return editLocale(id, { name, isDefault }).then(onClose);
+    return editLocale(id, { name, isDefault })
+      .then(() => {
+        console.log('ici');
+        setShouldUpdateMenu(true);
+      })
+      .then(onClose);
+  };
+
+  const handleClose = () => {
+    if (shouldUpdateMenu) {
+      updateMenu();
```

Comment:
I think we should just call the update menu when closing no?

Reply:
Well I think it's better to do it once the modal is closed so it has the same behaviour as the other modals in the app.

Reply:
But you will always enter the handleClose modal when the modal closes no? Why to put an intermediate state then? 🤔

Reply:
Because we don't want the menu to be regenerated when no modification has been made.

Reply:
If that's the case, WYT of using a ref instead? It would prevent from rerendering then